frame/support/procedural/src/pallet/parse/storage.rs

```diff
@@ -25,13 +25,26 @@ mod keyword {
 	syn::custom_keyword!(Error);
 	syn::custom_keyword!(pallet);
 	syn::custom_keyword!(getter);
+	syn::custom_keyword!(storage_name);
 	syn::custom_keyword!(OptionQuery);
 	syn::custom_keyword!(ValueQuery);
 }
 
-/// Parse for `#[pallet::getter(fn dummy)]`
-pub struct PalletStorageAttr {
-	getter: syn::Ident,
+/// Parse for one of the following:
+/// * `#[pallet::getter(fn dummy)]`
+/// * `#[pallet::storage_name = "CustomName"]`
```

Comment:
```suggestion
/// * `#[pallet::storage_prefix = "CustomName"]`
```

I might be splitting hairs here, but I think prefix is really more descriptive here. What do you think?

Reply:
Yeah, I think this makes more sense. I was worried about how we educate our users to differentiate between names and prefixes, but I realized that the term "storage prefix" is ubiquitous everywhere and I don't think we ever used "storage name" in our docs, so let's go ahead and rename this.